### ISSUE TITLE
overfitting

### ISSUE THREAD

**Issue by zpmmehrdad** [AlexeyAB/darknet#92]
Hi,
I have 67 classes and for each of the classes respectively :
class no 0 => 3909 images
class no 1 => 2087 images
class no 2 => 1875 images
class no 3 => 1530 images
class no 4 => 1366 images
.
.
.
class no 65 => 15 images
class no 66 => 14 images
class no 67 => 11 images

I use yolov2.cfg 
max_batches = 120000
filters = 360
classes = 67
and after 18000 iteration avg is 1.8 and will not increase.
Does the over-fitting happen? where is the problem?
Thanks.

**Comment by OPPOA113** [comment]
@AlexeyAB @zpmmehrdad 
hi.
i also have this problem.
i have 4 classes : 
0-->1492 images
1-->6730 images
2-->798 images
3-->968 images

my iteration avg is about 2.1 and will not increase too.
i have try to change the learning rate, but it did not work.

**Comment by AlexeyAB** [comment]
@zpmmehrdad @OPPOA113 Hi,

* AVG should decrease (should not increase) during training. 
* AVG does not have to have any specific low value. AVG just has to stop decreasing.
* And yes, if avg will not increase during training ~1000 iterations, then you should stop, and do `recall` for last weights and some of previous weights: https://github.com/AlexeyAB/darknet#when-should-i-stop-training

The best way, if all of your classes have the same number of images. If they differ more than twice, you can simply duplicate some images and label files (and add them to train.txt).

**Comment by zpmmehrdad** [comment]
Hi @AlexeyAB 
yes AVG should decrease.i made a mistake
that's mean you are saying add same path to train.txt for increase number of images.the over-fitting does not happen. 
how many images per class are sufficient?

**Comment by AlexeyAB** [comment]
@zpmmehrdad So should be ~2000 images per class.
Yes, you can simply duplicate some of images in train.txt. But much better if you will use really 2000 different images for each class.

**Comment by zpmmehrdad** [comment]
@AlexeyAB thanks
I have to change max_batches.
Max=number of classes * 2000
Is it right?

**Comment by AlexeyAB** [comment]
@zpmmehrdad Yes.

**Comment by vg123** [comment]
I have 120 images in training file and I have only one class to detect. I pu max_batches =2000
But It is giving segmentation fault after 1400th iteration, and when I changed ratio of batch to subdivisions 1( 16,16).  it is getting killed after 120 iterations .

please tell me, what is the problem?

**Comment by AlexeyAB** [comment]
@vg123 Try to re-check your cfg-file and to update your source code from: https://github.com/pjreddie/darknet
